core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -827,9 +868,21 @@ class DAGScheduler(
     // might modify state of objects referenced in their closures. This is necessary in Hadoop
     // where the JobConf/Configuration object is not thread-safe.
     var taskBinary: Broadcast[Array[Byte]] = null
+
+    // Check if RDD serialization debugging is enabled
+    val debugSerialization: Boolean = sc.getConf.getBoolean("spark.serializer.debug", false)
```

Comment:
Ah I see - this does that already. Yeah so I'd just remove the config option and just always print debugging output if there is a failure. We usually try not to add config options unless there is a really compelling reason to not have the feature enabled.